### PR TITLE
Pull request of Sudoku Solver using CNN

### PR DESCRIPTION
**Pull request by keshripritesh**
### ✅ Pull Request Title:

**Add Sudoku Solver using Convolutional Neural Networks**

---

### 📌 Description

This PR introduces a deep learning-based **Sudoku Solver** which uses **Convolutional Neural Networks (CNNs)** to recognize and solve Sudoku puzzles. This project is implemented in Python using **TensorFlow** and **Keras**.

---

### 🧠 Version

```
v1.1.2
```

---

### 🎯 Goal

To develop an intelligent system capable of reading a Sudoku puzzle from an image and predicting the solution using deep learning, specifically convolutional neural networks.

---

### 📁 Project Contents

| File           | Description                                            |
| -------------- | ------------------------------------------------------ |
| `sudoku.ipynb` | Jupyter notebook to test and run the model             |
| `model.py`     | CNN architecture and model training implementation     |
| `report.pdf`   | Detailed project explanation with implementation steps |

---

### 🚀 Performance

* ✅ Achieved **\~95% accuracy** on Sudoku puzzle solving
* 📷 Image input converted into grid format and predicted via CNN
* 🧠 Efficiently handles multiple puzzle inputs in real-time

---

### 🔗 Dataset

Used dataset:
📦 [[Kaggle - Sudoku Dataset by Bryan Park](https://www.kaggle.com/bryanpark/sudoku)](https://www.kaggle.com/bryanpark/sudoku)

---

### 📦 Requirements

Make sure you have the following dependencies installed:

```bash
python==3.7
tensorflow==2.2.0
keras==2.4.3
```

You can include them in a `requirements.txt` like this:

```
tensorflow==2.2.0
keras==2.4.3
```

---

### 👨‍💻 How It Works (Quick Overview)

1. Puzzle images are processed into a 9x9 grid
2. A CNN model classifies digits (1–9)
3. Model predicts missing digits in the Sudoku
4. Final grid is displayed with the complete solution

---

### 📸 Sample Output

```plaintext
Input Sudoku:
[5, 3, 0, 0, 7, 0, 0, 0, 0]
...

Predicted Output:
[5, 3, 4, 6, 7, 8, 9, 1, 2]
...
```

---

### ✅ Checklist

* [x] Model and training code provided
* [x] Sample notebook to run predictions
* [x] Dataset linked
* [x] Project documentation (`report.pdf`) included
* [x] Accuracy details and usage explained

---

### CODE REVIEW

**Comment by PRIYANSHU2026** [comment]
@keshripritesh  Good Work this is Very New
